velox/exec/tests/PrintPlanWithStatsTest.cpp

```diff
@@ -375,3 +376,52 @@ TEST_F(PrintPlanWithStatsTest, tableWriterWithTableScan) {
        {"        totalRemainingFilterTime\\s+sum: .+, count: .+, min: .+, max: .+"},
        {"        totalScanTime    [ ]* sum: .+, count: .+, min: .+, max: .+"}});
 }
+
+TEST_F(PrintPlanWithStatsTest, taskAPI) {
+  // Test nullptr PlanNode.
+  core::PlanFragment planFragment;
+  auto taskWithNoPlanNode = exec::Task::create(
```

Comment:
Should Task::create API fail if there is no plan? Are we missing such check?

Reply:
There is no check currently. We can add a check and remove planNode checks everywhere.
I was not sure if there was a such use case inside Meta. Looks like not.

Reply:
> We can add a check in Task::create and remove planNode checks everywhere.

That would be nice. I assume Task created with no plan is not usable.

Reply:
Should `struct PlanFragment` have this check instead?